### ISSUE TITLE
Error "Compiler reserves name 'require'" doesn't make sense for JS

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#50132]
```js
    const someModule = require("google-closure-compiler");
    /**
     * @param {string} str
     * @return {number}
     */
    function require(str) { return str.length }
             ~~~~~~~
!!! error TS2441: Duplicate identifier 'require'. Compiler reserves name 'require' in top level scope of a module.
    export {someModule};
```

Actual: Error to do with emitting a module as commonjs
Expected: No error in a .js file, which shouldn't undergo transformation for emit.

**Comment by sandersn** [comment]
Never mind, @weswigham reminded me that .js files definitely undergo module transformation, so this error is correct.
